Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -37,6 +37,28 @@ TimeSplitter::TimeSplitter(const DateAndTime &start, const DateAndTime &stop) {
   clearAndReplace(start, stop, DEFAULT_VALUE);
 }
 
+/**
+ * Note: The amount of X values in input MatrixWorkspace must be 1 larger than the amount of Y values.
+ * There are NO undefined split regions here.
+ **/
+TimeSplitter::TimeSplitter(const Mantid::API::MatrixWorkspace_sptr &ws) {
+  if (ws->getNumberHistograms() != 1) {
+    throw std::runtime_error("MatrixWorkspace can only have 1 histogram when constructing TimeSplitter.");
+  }
+
+  const auto X = ws->binEdges(0);
```

Comment:
according to the docstring, there should be a validation here that all values in `X` should be equal or greater than zero, otherwise throw a `RuntimeError`